src/stactools/core/addasset.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+from pystac import Asset, Item
+from pystac.utils import is_absolute_href, make_relative_href
+
+from stactools.core.copy import move_asset_file_to_item
+
+logger = logging.getLogger(__name__)
+
+
+def add_asset_to_item(
+    item: Item,
+    key: str,
+    asset: Asset,
+    move_assets: bool = False,
+    ignore_conflicts: bool = False,
+) -> Item:
+    """Adds an asset to an item.
+
+    Args:
+        item (Item): The PySTAC Item to which the asset will be added.
+        key (str): The unique key of the asset.
+        asset (Asset): The PySTAC Asset to add.
+        move_assets (bool): If True, move the asset file alongside the target item.
+        ignore_conflicts (bool): If True, asset with the same key will not be added,
+            and asset file that would overwrite an existing file will not be moved.
+            If False, either of these situations will throw an error.
+
+    Returns:
+        Item: Returns an updated Item with the added Asset.
+            This operation mutates the Item.
+    """
+    item_href = item.get_self_href()
+    asset_href = asset.get_absolute_href()
+    if key in item.assets:
+        if not ignore_conflicts:
+            raise Exception(
+                "Target item {} already has asset with key {}, "
+                "cannot add asset in from {}".format(item, key, asset_href)
+            )
+    else:
+        if asset_href is None:
+            raise ValueError(
+                f"Asset {repr(asset)} must have an absolute HREF to be added"
+            )
+        if item_href is None and (move_assets or not is_absolute_href(asset.href)):
+            # An absolute href for the item is required if we need to move the
+            # asset into the item or if the asset href is relative
+            raise ValueError(
+                f"Target Item {repr(item)} must have an HREF to add an asset"
+            )
```

Comment:
Syntax isn't perfect but you get the idea

```suggestion
        if item_href is None and move_assets:
            # An absolute href for the item is required if we need to move the
            # asset into the item or if the asset href is relative
            raise ValueError(
                f"Target Item {repr(item)} must have an HREF to move an asset to the item"
            )
        elif item_href is None and not is_absolute_href(asset.href):
            raise ValueError(f"Target Item {repr(item)} must have an href to add an asset with a relative href")
```